src/main/java/core/basesyntax/figure/Circle.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax.figure;
+
+import core.basesyntax.Figure;
+import core.basesyntax.IShape;
+
+public class Circle extends Figure implements IShape {
+    private double radius;
+
+    public Circle(double radius, String name) {
+        this.radius = radius;
+        setName(name);
+    }
+
+    @Override
+    public double getArea() {
+        return Math.PI * radius * radius;
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Figure: " + this.getName()
+                + ", area: " + this.getArea()
+                + " sq. units, radius: " + this.radius
+                + " units, color: " + this.getColor());
```

Comment:
do you need `this` keyword here?

same for all other implementations of `draw()` method in child classes